hrcase/CRM/HRCase/Upgrader.php

```diff
@@ -116,14 +116,26 @@ public function upgrade_1402() {
     $this->up1402_removeUnusedCaseTypes();
     //Removes CiviCase activity types which should belong to CiviTask component
     $this->removeActivityTypesList($defaultActivityTypes, 'CiviCase');
-    $this->up1402_createDefaultCaseTypes($defaultCaseTypes);
+    $this->createOrUpdateDefaultCaseTypes($defaultCaseTypes);
     $this->up1402_createDefaultActivityTypes($defaultActivityTypes);
     $this->changeActivityTypeComponent('Open Case', 'CiviCase', 'CiviTask');
 
     return TRUE;
   }
 
   /**
+   * Resets all default case types discarding any customization to match new
+   * activity workflow
+   */
+  public function upgrade_1404() {
+    $this->installActivityTypes('CiviDocument', ['P45']);
```

Comment:
Good point about the duplicate functions, and I've changed them to use the existing ones. To avoid duplication I changed the structure of the `$defaultActivityTypes`. This meant changing some methods in the Upgrader class too.

What is your opinion on using the `up1402_` style of naming methods in upgraders? I think it makes it harder to find duplicate functions if the name has a unique code in it.